packages/codegen-ui/lib/utils/form-to-component/map-form-definition-to-component.ts

```diff
@@ -98,55 +98,84 @@ const ctaButtonConfig = (): StudioComponentChild => {
         value: '1rem',
       },
     },
-    children: [
-      {
-        componentType: 'Button',
-        name: 'CancelButton',
-        properties: {
-          children: {
-            value: 'Cancel',
-          },
-          type: {
-            value: 'button',
-          },
+    children: [],
+  };
+
+  if (formDefinition.buttons.buttonConfigs.cancel) {
+    CTAComponent.children?.push({
+      componentType: 'Button',
+      name: 'CancelButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.cancel.props.children,
+        },
+        type: {
+          value: 'button',
         },
       },
-      {
-        componentType: 'Flex',
-        name: 'SubmitAndResetFlex',
-        properties: {},
-        children: [
-          {
-            componentType: 'Button',
-            name: 'ClearButton',
-            properties: {
-              children: {
-                value: 'Clear',
-              },
-              type: {
-                value: 'reset',
-              },
-            },
-          },
-          {
-            componentType: 'Button',
-            name: 'SubmitButton',
-            properties: {
-              children: {
-                value: 'Submit',
-              },
-              type: {
-                value: 'submit',
-              },
-              variation: {
-                value: 'primary',
-              },
-            },
-          },
-        ],
-      },
-    ],
+    });
+  }
+
+  const rightAlignCTA: StudioComponentChild = {
+    componentType: 'Flex',
+    name: 'SubmitAndResetFlex',
+    properties: {},
+    children: [],
   };
+
+  if (formDefinition.buttons.buttonConfigs.clear && rightAlignCTA.children) {
+    rightAlignCTA.children.push({
+      componentType: 'Button',
+      name: 'ClearButton',
+      properties: {
+        children: {
+          value: formDefinition.buttons.buttonConfigs.clear.props.children,
+        },
+        type: {
+          value: 'reset',
+        },
+      },
+    });
+  }
+
+  if (formDefinition.buttons.buttonConfigs.submit) {
+    rightAlignCTA.children?.push({
+      componentType: 'Button',
+      name: 'SubmitButton',
+      properties: {
+        children: {
```

Comment:
The advantage of having `props` on the form definition element is that you don't have to hand map like this. If you look at how the other elements are mapped, you can see. Also, componentType and name can be mapped in a DRY way too. For name, you can do a simple string conversion or mapping.

For type and variation, you can also map this from the form definition element. ('variant' should be 'variation'. My bad)